pkg/nvmeclient/nvme_client.go

```diff
@@ -495,6 +497,26 @@ func ConnectAll(discoveryRequest *hostapi.DiscoverRequest, maxIOQueues int) ([]*
 	return ctrls, nil
 }
 
+func connectNVMEDevicesWithRetry(request *ConnectRequest) (*CtrlIdentifier, error) {
+	var err error
+	var ctrlID *CtrlIdentifier
+	attempts := uint(5)
+	delay := time.Millisecond * 10
+	retry.DefaultDelayType = retry.BackOffDelay
+
+	err = retry.Do(func() error {
+		ctrlID, err = Connect(request)
+		if err == nil {
+			return nil
+		}
+
+		return err
+	}, retry.Attempts(attempts),
+		retry.Delay(delay))
```

Comment:
10 msec doesn't sound like a lot, unless we're counting on the exponential delay? What are the expected delays?

Reply:
yes I used `BackOffDelay` - `is a DelayType which increases delay between consecutive retries`
that means that we have 5 retries with the following delays: 10, 20, 40, 80, 160 (ms)